packages/template/src/lib/stack-app.ts

```diff
@@ -129,13 +140,50 @@ function getDefaultSuperSecretAdminKey() {
   return process.env.STACK_SUPER_SECRET_ADMIN_KEY || throwErr(new Error("No super secret admin key provided. Please copy your key from the Stack dashboard and put it in the STACK_SUPER_SECRET_ADMIN_KEY environment variable."));
 }
 
-function getDefaultBaseUrl() {
-  const url = getPublicEnvVar("NEXT_PUBLIC_STACK_API_URL") || getPublicEnvVar("NEXT_PUBLIC_STACK_URL") || defaultBaseUrl;
+/**
+ * Returns the base URL for the Stack API.
+ *
+ * The URL can be specified in several ways, in order of precedence:
+ * 1. Directly through userSpecifiedBaseUrl parameter as string or browser/server object
```

Comment:
by definition if you specify an argument with the return value it's not a "default" anymore

why not just `truncateTrailingSlash(baseUrl || getDefaultBaseUrl())`?